app/geo/gui/src/ts/twxplore/gui/geo/reducers/map/mapReducer.ts

```diff
@@ -395,8 +394,77 @@ export const mapReducer = (state: MapState, action: BaseAction): MapState => {
 
       break;
     }
+
+    //THE MapFeatureTypeState.NEEDS_POPUP_CHANGE and MapFeatureTypeState.NEEDS_LNG_AND_LAT dispatch actions that use this case
+    case "@@kepler.gl/LAYER_CONFIG_CHANGE": {
+      const layerConfigChangeAction: any = action;
+      if (
+        layerConfigChangeAction.oldLayer === undefined ||
+        layerConfigChangeAction.newConfig === undefined
+      )
+        break;
+      const layerIdOfFeatureType: string =
+        layerConfigChangeAction.oldLayer.config.dataId;
+      let featuresByTypeOfFeatureType =
+        result.featuresByType[layerIdOfFeatureType];
+
+      /*The layerConfigChangeAction is used for several different purposes (such as changing layer visibility), 
+      the below code is to check if we're currently in this case because the label of the layer is being changed. If so, that is our signal
+      that we are in the NEEDS_POPUP_CHANGE case and it is time to move forward and change the state */
+      if (Object.keys(layerConfigChangeAction.newConfig).includes("label")) {
+        //If this layer is attached to the 'Transmission' FeatureType... then next step is to change the layer. Else, we're done.
+        if (
+          FeatureType[layerIdOfFeatureType as keyof typeof FeatureType] ===
+          FeatureType.Transmission
+        ) {
+          featuresByTypeOfFeatureType.featureTypeState =
+            MapFeatureTypeState.NEEDS_LAYER_CHANGE;
+        } else {
+          featuresByTypeOfFeatureType.featureTypeState =
+            MapFeatureTypeState.FINISHED_SETUP;
+        }
+      }
+      /*Checking to see if the action was used to change the 'columns' of the layer. If so, then it signifies 
+      that we are in MapFeatureTypeState.NEEDS_LNG_AND_LAT and it is time to move to NEEDS_3D_ENABLED*/
+      if (Object.keys(layerConfigChangeAction.newConfig).includes("columns")) {
+        result.featuresByType[FeatureType.Transmission].featureTypeState =
```

Comment:
Hard-coding a FeatureType here smells. This should be abstracted into the strategy.